### PR TITLE
Add workflow_dispatch github actions trigger.

### PR DESCRIPTION
**Pull request by philmcmahon**
## What does this change?
Adds the [workflow_dispatch](https://github.blog/changelog/2020-07-06-github-actions-manual-triggers-with-workflow_dispatch/) trigger to make it possible to manually run CI/CD

## How to test

Go into the [github actions console](https://github.com/guardian/editions-card-builder/actions) then select a branch and run the action on that branch.

## How can we measure success?
Being able to deploy arbitrary branches to PROD, re run CI/CD without having to push code changes. 

## Have we considered potential risks?
It makes it easier to put a development branch on PROD, so should be used with caution. However, this behaviour is already possible via the github api so let's put it in a nice UI!

### CODE REVIEW

File: .github/workflows/cd.yml
Comment (akash1810) [inline .github/workflows/cd.yml:3]: Not sure this'll have much benefit over the current scheme of running on `push`. I think this change will mean we can deploy what's currently deployed, on demand.

Comment (jennygrahamjones) [reply .github/workflows/cd.yml:3]: Does the current configuration allow us to  deploy _any_ branch?

Comment (akash1810) [reply .github/workflows/cd.yml:3]: Re-read the PR description and it sounds like we want to be able to deploy non-default branches to PROD? Any reason why, as this doesn't follow the majority of other apps. I think we'd want to tweak the [branches rule](https://github.com/guardian/editions-card-builder/blob/8ac482ece01cd92f91e6ac07b49c1038ca133245/.github/workflows/cd.yml#L3-L5) too.

Comment (philmcmahon) [reply .github/workflows/cd.yml:3]: I suppose in the absence of a CODE stack it can be useful to deploy something to PROD from a development branch for temporary testing. The use case I came across was that I wanted Katy to test the new cover card crop sizes, but if it didn't work I needed to be able to revert quickly. Perhaps the ideal would have been:

 - get approval for change, merge
 - test, it doesn't work
 - do revert of PR
 - (wait for someone to approve revert)
 - merge revert

but the testing was taking place fairly early in the morning when people had standups etc, so hard to guarentee that  someone will be around. With this change you could:

- deploy dev branch to PROD
- test, it doesn't work
- deploy master back to PROD

Arguably, we should just do our testing in a time when there's support from others to review PRs!  So happy for this to be closed, I think it's a useful functionality to have though even if it's a "use with caution" thing

Comment (philmcmahon) [reply .github/workflows/cd.yml:3]: I guess the real solution here would be to have a CODE environment, but that's more trouble than it's worth!

Comment (jennygrahamjones) [reply .github/workflows/cd.yml:3]: >  I guess the real solution here would be to have a CODE environment, but that's more trouble than it's worth!

In lieu of that, this feels like it simplifies the testing workflow and makes it clearer when we are actually testing something from a particular branch, as opposed to potentially polluting the commit history with a number of merges and reverts.

Comment (philmcmahon) [reply .github/workflows/cd.yml:3]: As someone who's run into the merge/revert cycle 3 times now I agree with this! I think this would be a useful change, but also happy to close this PR if it doesn't seem like a good idea.

Comment (jennygrahamjones) [reply .github/workflows/cd.yml:3]: I think this is a useful change to make. If we merge it, it might be useful to have something in the README about how to use this approach.

**Review by jennygrahamjones — APPROVED** [review]
Approving as it seems we are unlikely to create a CODE stack for this any time soon, and there have been a number of PRs that would have benefitted from an easier test process.

I do still think it would be handy to mention this as a feature in the README.

**Review by akash1810 — APPROVED** [review]
Let's merge this to unblock ourselves. Having a proper CODE environment is still the preferred solution IMO, though a lot more work.

**Pull request merged**
